benchmark/README.md

```diff
@@ -49,79 +107,151 @@ The following build options are available:
+current benchmark results will be compared to the previous run's results if
+available. Results for each benchmark run are logged for future comparison.
+
+For branch based development, take a baseline benchmark on the Swift `master`
+branch, switch to a development branch containing potentially performance
+impacting changes, and run the benchmarks again. Upon benchmark completion, the
+benchmark results for the development branch will be compared to the most
+recent benchmark results for `master`.
+
+### Using standalone + swiftpm
+
+When building with SwiftPM, we only build a single binary of the low level
+driver (called SwiftBench) rather than compile one low level driver for each
+individual optimization setting. So it is important to make sure that if one is
+building in Xcode that one is building in release mode.
+
+In the case of building with swiftpm, the file will be in the directory .build
+in the directory where swift build was invoked. If one uses Xcode, then
+SwiftBench will be a listed build product in Xcode. To find the path of
+SwiftBench easily, drag and drop the product from the Xcode window in
+terminal. Then one should be able to invoke SwiftBench with the relevant options
+without problem. E.g.:
+
+```
+/Users/gottesmm/Library/Developer/Xcode/DerivedData/swiftbench-xxxxxxxxxx/Build/Products/Release/SwiftBench --num-samples 3
+```
+
+### Using standalone + cmake
 
-### Usage
+Run:
 
-`./Driver [ test_name [ test_name ] ] [ option [ option ] ]`
+```./bin/Benchmark_Driver run -i 3```
+
+### Low Level Driver Usage
+
+`./Benchmark_{O,Onone,Osize} [option] [test_names]`
+
+If test names is empty, then it is assumed that all tests should be run. If the
+number of samples is not specified then it is assumed that a sample of 1 is
+wanted.
+
+Options:
 
-* `--num-iters`
-    * Control the number of loop iterations in each test sample
 * `--num-samples`
-    * Control the number of samples to take for each test
+    * Control the number of samples to take for each test.
 * `--list`
     * Print a list of available tests
+* `--help`
+    * List available options.
+* `--tags=`
+    * Only run benchmarks associated with the given regex of tags.
+* `--skip-tags=`
+    * Do not skip benchmarks that match the given regex of tags. To run all
+      benchmarks including unstable benchmarks, pass in ``*``, e.g.:
+      `--skip-tags=*`
 
-### Examples
-
-* `$ ./Benchmark_O --num-iters=1 --num-samples=1`
-* `$ ./Benchmark_Onone --list`
-* `$ ./Benchmark_Osize Ackermann`
+Examples:
 
-### Note
-As a shortcut, you can also refer to benchmarks by their ordinal numbers.
-The regular `--list` option does not provide these, but you can run:
-* `$ ./Benchmark_O --list --run-all | tail -n +2 | nl`
-You can use ordinal numbers instead of test names like this:
-* `$ ./Benchmark_O 1 42`
-* `$ ./Benchmark_Driver run 1 42`
+```
+    $ # Run all non-unstable benchmarks for 3 samples each and compute summary statistics.
```

Comment:
Non-unstable? As in stable? Or maybe “Run all benchmarks (except those tagged `.unstable`) for 3 samples…”

Reply:
I am also fine with this.